src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -217,9 +265,21 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 	{
 		/*
 		 * We are planning a call to a distributed function within a Citus backend,
-		 * that means that this is the delegated call.
+		 * that means that this is the delegated call. If the function is forcefully
+		 * delegated, capture the distribution argument.
+		 */
+		CheckDelegatedFunctionExecution(procedure, funcExpr);
+
+		/*
+		 * InSafeDelegatedFunctionCall flag grants the levy to do remote tasks from
+		 * a delegated function, we can do that only if the function is delegated
+		 * without an explicit transaction on the client session.
 		 */
-		InDelegatedFunctionCall = true;
+		if (!IsMultiStatementTransaction())
```

Comment:
could we look at the force flag here to be a little more explicit? I still don't like that we are inferring the fact that we are doing a regular delegated function call from something seemingly unrelated.

Reply:
Out of the four combinations, only `Force + Begin` is not safe, I made it as explicit as possible.